pulp_ansible/app/serializers.py

```diff
@@ -381,9 +381,9 @@ def validate(self, data):
             data = self.deferred_validate(data)
 
         sha256 = data["file"].hashers["sha256"].hexdigest()
-        artifact = Artifact.objects.filter(sha256=sha256).first()
-        if artifact:
-            ValidationError(_("Artifact already exists"))
+        cv = CollectionVersion.objects.filter(sha256=sha256).first()
+        if cv:
+            raise ValidationError(_("Collection Version already exists"))
```

Comment:
Another story for another day, but i think we need this serializer to parse the artifact in deferred_validate.

Reply:
I think you are right, we should write plugin documentation for uploading content to Pulp for when and where fields should be processed and validated because I can never remember off the top of my head.